scripts/openstack-loopback-lvm

```diff
@@ -1,45 +1,73 @@
 #!/bin/sh
 
+function truncate_file()
+{
+    local f=$1
+    # calc wanted size
+    size=$(df -P -k $(dirname $f)|tail -1| \
+                  perl -ne 'm/^\S+\s*\d+\s+\d+\s+(\d+)/; print int($1*0.3)')
```

Comment:
That regular expresssion is hard to read, and since it only plucks out a single field, something like this might be easier on the eyes:

`perl -ane 'print int($F[3] * 0.3)'`